### ISSUE TITLE
Problems at OpenSSL test stage

### ISSUE THREAD

**Issue by scottaki** [ab77/netflix-proxy#109]
Similar in nature to #108 but potentially different enough to warrant another issue. Have followed new readme steps and have no problems with build until the stage after the DNS Test – the OpenSSL test – which fails without explanation (and retries only to fail again). Have looked at syslog with nothing useful revealed. Running Ubuntu 14.04, and had no issues with the new-auth version before the merge into master.

After test stage begins, it seems to say "Connected", but then fails following this message. Nothing else outputted to the command line. Any suggestions for places to look for further information relating to this issue? May be simply a configuration issue, but just a little stuck for next-steps now.

**Comment by ab77** [comment]
Try now, I've fixed the issue with BIND container not starting (restart loop due to duplictae zone sin `zones.override` file). Tested on DO and Vultr OK.

**Comment by scottaki** [comment]
Thanks! ...but didn't work for me unfortunately. Seems like there might be multiple issues (at least with my configuration). Those fixes (or last few additions) to `zones.override` seemed to be looking for `/data/db.override` instead of `/data/conf/db.override`. Changing this to `/data/conf/db.override` seemed to fix the bind issue, but OpenSSL tests still failing.

Logs from the sniproxy container read:

```
Setting sniproxy resolver to 172.17.0.3
Setting sniproxy resolver to 172.17.0.4
2016-03-24 10:03:12 recv(): Connection timed out, closing connection
2016-03-24 10:03:32 recv(): Connection timed out, closing connection
```

Any thoughts or suggestions?

**Comment by scottaki** [comment]
My VPS was pre-configured with a block of IPv6 addresses, so not sure if that's now complicating the issue because of the new IPv4 to IPv6 tunnel...

**Comment by ab77** [comment]
Build your VPS without IPv6 support, or disable it after build (remove the IPv6 addrs).

Then configure the tunnel, modify network interfaces file and `ifup he-ipv6`.

Then `ping6 google.com`. Don't go any further until you get that working..

**Comment by ab77** [comment]
(fixed `zones.override` cheers for spotting)

**Comment by scottaki** [comment]
It's really strange... IPv6 seems to be working. `ping6 google.com` works, and `traceroute6` shows that route is using the tunnel correctly. However, the SSL negotiation is failing. If I pull that test command out of the build.sh file and call it directly, with a few extra verbose options enabled, I get this:

```
GET HTTP/1.1 | timeout 10 openssl s_client -CApath /etc/ssl/certs -servername netflix.com -connect ${EXTIP}:443 -msg -state -no_ticket -no_tls1
CONNECTED(00000003)
SSL_connect:before/connect initialisation
>>> TLS 1.2 Handshake [length 0132], ClientHello
SSL_connect:unknown state
```

Looking at netstat, it doesn't seem like there's anything listening on tcp:443 however... should there be? The docker-proxy is listening on tcp6:443 and tcp6:80, but nothing tcp(4). I can't see how the request can go anywhere with nothing listening on ${EXTIP}:443, but I don't understand the setup here well enough to know for sure (to be completely honest).

```
tcp6       0      0 :::80                   :::*                    LISTEN      2932/docker-proxy
tcp6       0      0 :::443                  :::*                    LISTEN      2911/docker-proxy
```

**Comment by scottaki** [comment]
OK. Having looked through the build.sh file in a little more detail, I see the PRREROUTING rules adding the redirect to port 8080 for those comments I made above. However again, Caddy is only listening on tcp6:8080. Looking at Caddyfile.template I thought that the SSL support on Caddy had been abandoned, so not sure what's happening with my config here...

**Comment by ab77** [comment]
Hmm, can you show me the output of the build script? Pastebin it or something..

**Comment by scottaki** [comment]
Sure thing: [](http://pastebin.com/cJ5FiDaR)

Seems like there are a stack of duplicates in `/etc/sysctl.conf` looking at the output, but everything seems to pass until that OpenSSL test stage. Really appreciate the help!

**Comment by ab77** [comment]
There the problem:
`DOCKER_OPTS='--iptables=false --ipv6 --fixed-cidr-v6=“DOCKER_IPv6_SUBNET/125"'`

Can you run `ifconfig eth0` as well as `ifconfig he-ipv6` and paste the output?

**Comment by scottaki** [comment]
Oh, sorry, I replaced that IPv6 address manually there... just saw an IP and went HIDE!

**Comment by scottaki** [comment]
ifconfig eth0
eth0      Link encap:Ethernet  HWaddr 00:16:3c:37:d8:9c  
          inet addr:VPS_IP  Bcast:192.161.48.127  Mask:255.255.255.192
          inet6 addr: fe80::216:3cff:fe37:d89c/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:79173 errors:0 dropped:13 overruns:0 frame:0
          TX packets:33082 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:109392821 (109.3 MB)  TX bytes:2794217 (2.7 MB)

ifconfig he-ipv6
he-ipv6   Link encap:IPv6-in-IPv4  
          inet6 addr: VPS_IP6/64 Scope:Global
          inet6 addr: fe80::c0a1:3073/64 Scope:Link
          UP POINTOPOINT RUNNING NOARP  MTU:1480  Metric:1
          RX packets:4170 errors:0 dropped:0 overruns:0 frame:0
          TX packets:2039 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:5949712 (5.9 MB)  TX bytes:180900 (180.9 KB)

**Comment by ab77** [comment]
Can you try rebuilding with `./build.sh -z 1`..

**Comment by scottaki** [comment]
WORKED! Amazing! Now I'll have to actually test it out and see if we're unblocked again. Any thoughts on why that was causing issues?

**Comment by ab77** [comment]
It's the caching-resolver.. It is very sensitive to IPv6 config differences between different providers. I'll default it to disable again..

**Comment by scottaki** [comment]
OK. Good to know! Thanks so much for the help!

**Comment by ab77** [comment]
Good stuff.

**Comment by scottaki** [comment]
Just to update (no need to reopen this issue): while the build works, everything appears to be OK, Netflix is reporting a general "internet connection issue" (M7031-1105). Will have to troubleshoot further on my end.
